src/main/java/net/mcreator/ui/dialogs/ListSelectorDialog.java

```diff
@@ -59,8 +66,17 @@ public ListSelectorDialog(MCreator mcreator, Function<Workspace, List<T>> entryP
 
 		JButton selectButton = L10N.button("dialog.item_selector.use_selected");
 		selectButton.addActionListener(e -> {
-			setVisible(false);
-			dispose();
+			T selectedValue = list.getSelectedValue();
+			if (selectedValue == null && customValue) {

```

Comment:
This forces the user to insert a custom value if nothing has been selected. Is this intentional?

Reply:
I've put this as there must be a value set. If nothing is selected, then default must be custom.

Reply:
There doesn't have to be a value set. In procedures, it'll go back to the "No entry selected" value. In other selectors (e.g. sound selectors in the block mod element) it'll keep the currently selected value

Reply:
Fixed, done.